code/game/objects/items/cookie_synth.dm

```diff
@@ -26,32 +26,45 @@
 	if(emagged)
 		toggle_toxins(user)
 
+/obj/item/weapon/cookiesynth/AltClick()
+	if(src.loc == usr)
+		synthesize(src, usr, TRUE, TRUE)
+
 /obj/item/weapon/cookiesynth/proc/toggle_toxins(mob/user)
 	toxin = !toxin
 	to_chat(user,"<span class='warning'>You [toxin ? "dis" : "en"]able the [src]'s reagent safety checker!</span>")
 
 /obj/item/weapon/cookiesynth/afterattack(atom/A, mob/user, proximity)
-	if(cooldown > world.time)
-		return
+	synthesize(A, user, proximity)
+
+/obj/item/weapon/cookiesynth/proc/synthesize(atom/A, mob/user, proximity, put_in_hands = FALSE)
 	if(!proximity)
 		return
-	if(!(istype(A, /obj/structure/table) || isturf(A)))
+	var/turf/T = get_turf(A)
+	if(T.density)
+		return
+	if(!(istype(A, /obj/structure/table) || isturf(A) || put_in_hands)) //if it's a table, a turf, or it's being activated via AltClick()
```

Comment:
You don't need to check `isturf(A)` because this is already covered by not being a table. All turfs are not tables.

Reply:
It's checking if:
It's neither a table, a turf, nor is the put_in_hand flag active.
NOT ( table OR turf )

Reply:
ah, I missed the paren